testsuite/MDAnalysisTests/utils/test_persistence.py

```diff
@@ -36,97 +38,136 @@
+    @pytest.fixture()
+    def universe():
+        return mda.Universe(PDB_small, PDB_small, PDB_small)
+    
+    @staticmethod
+    @pytest.fixture()
+    def universe_n():
+        return mda.Universe(PDB_small, PDB_small, PDB_small, anchor_name="test1")
+    
+    @staticmethod
+    @pytest.fixture()
+    def ag(universe):
+        return universe.atoms[:20]
+    
+    @staticmethod
+    @pytest.fixture()
+    def ag_n(universe_n):
+        return universe_n.atoms[:10]
+    
+    @staticmethod
+    @pytest.fixture()
+    def pickle_str(ag):
+        return cPickle.dumps(ag, protocol=cPickle.HIGHEST_PROTOCOL)
+    
+    @staticmethod
+    @pytest.fixture()
+    def pickle_str_n(ag_n):
+        return cPickle.dumps(ag_n, protocol=cPickle.HIGHEST_PROTOCOL)
+
+    
+
+    # def setUp(self):
+    #     
+    #     self.universe = mda.Universe(PDB_small, PDB_small, PDB_small)
+    #     self.universe_n = mda.Universe(PDB_small, PDB_small, PDB_small,
+    #                                    anchor_name="test1")
+    #     self.ag = self.universe.atoms[:20]  # prototypical AtomGroup
+    #     self.ag_n = self.universe_n.atoms[:10]
+    #     self.pickle_str = cPickle.dumps(self.ag,
+    #                                     protocol=cPickle.HIGHEST_PROTOCOL)
+    #     self.pickle_str_n = cPickle.dumps(self.ag_n,
+    #                                       protocol=cPickle.HIGHEST_PROTOCOL)
+
+    # def tearDown(self):
+    #     # Some tests might delete attributes.
+    #     try:
+    #         del self.universe
+    #         del self.universe_n
+    #         del self.ag
+    #         del self.ag_n
+    #     except AttributeError:
+    #         pass
+
+    def test_unpickle(self, pickle_str, ag, universe):
         """Test that an AtomGroup can be unpickled (Issue 293)"""
-        newag = cPickle.loads(self.pickle_str)
+        newag = cPickle.loads(pickle_str)
         # Can unpickle
-        assert_array_equal(self.ag.indices, newag.indices)
-        assert_(newag.universe is self.universe,
+        assert_array_equal(ag.indices, newag.indices)
+        assert_(newag.universe is universe,
                 "Unpickled AtomGroup on wrong Universe.")
 
-    def test_unpickle_named(self):
+    def test_unpickle_named(self, pickle_str_n, ag_n, universe_n):
         """Test that an AtomGroup can be unpickled (Issue 293)"""
-        newag = cPickle.loads(self.pickle_str_n)
+        newag = cPickle.loads(pickle_str_n)
         # Can unpickle
-        assert_array_equal(self.ag_n.indices, newag.indices)
-        assert_(newag.universe is self.universe_n,
+        assert_array_equal(ag_n.indices, newag.indices)
+        assert_(newag.universe is universe_n,
                 "Unpickled AtomGroup on wrong Universe.")
 
-    def test_unpickle_missing(self):
+    def test_unpickle_missing(self, pickle_str_n, ag_n, universe_n):
         # Kill AtomGroup and Universe
-        del self.ag_n
-        del self.universe_n
+        del ag_n
```

Comment:
I don't think that `del` makes sense with these fixtures. 

Try not to pass them as arguments and leave out the `del` lines. Maybe they never appear... but in this case, I have no idea what the deep pytest magic actually does.